packages/@ember/-internals/runtime/lib/system/array_proxy.js

```diff
@@ -124,6 +124,16 @@ export default class ArrayProxy extends EmberObject {
     this._removeArrangedContentArrayObserver();
   }
 
+  [CUSTOM_TAG_FOR](key) {
+    if (key === '[]') {
```

Comment:
is 'length' not needed?

Reply:
This is the tag that entangled by `get` and `#each`, so I don't think so. If `length` is entangled, that means user code has definitely entangled it by accessing it directly, so it'll already have updated (since it's a real property).

Reply:
Ended up adding it since we need to entangle the tag anyways.